### PR TITLE
Add CredentialName and GoogleCloud fields in ClusterUserAuth spec

### PR DESCRIPTION
**Pull request by Shaad7**
Signed-off-by: Masudur Rahman <masud@appscode.com>

### CODE REVIEW

File: apis/cluster/v1alpha1/cluster_user_auth_types.go [hunk 1]
```diff
@@ -84,9 +84,16 @@ type ClusterUserAuthSpec struct {
...
 	// Provider Access Token params
 	// +optional
-	Provider    TokenProviderName    `json:"provider,omitempty" protobuf:"bytes,12,opt,name=provider,casttype=TokenProviderName"`
-	GoogleOAuth *GoogleOAuthProvider `json:"googleOAuth,omitempty" protobuf:"bytes,13,opt,name=googleOAuth"`
-	AWS         *AWSProvider         `json:"aws,omitempty" protobuf:"bytes,14,opt,name=aws"`
+	Provider       TokenProviderName      `json:"provider,omitempty" protobuf:"bytes,12,opt,name=provider,casttype=TokenProviderName"`
+	GoogleOAuth    *GoogleOAuthProvider   `json:"googleOAuth,omitempty" protobuf:"bytes,13,opt,name=googleOAuth"`
+	AWS            *AWSProvider           `json:"aws,omitempty" protobuf:"bytes,14,opt,name=aws"`
+	CredentialName string                 `json:"credentialName,omitempty" protobuf:"bytes,15,opt,name=credentialName"`
+	GoogleCloud    *GoogleCloudCredential `json:"googleCloud,omitempty" protobuf:"bytes,16,opt,name=googleCloud"`
```
Comment (masudur-rahman) [inline apis/cluster/v1alpha1/cluster_user_auth_types.go:91]: Keep both google credential side by side

File: apis/cluster/v1alpha1/cluster_user_auth_types.go [hunk 2]
```diff
@@ -84,9 +84,16 @@ type ClusterUserAuthSpec struct {
 
 	// Provider Access Token params
 	// +optional
-	Provider    TokenProviderName    `json:"provider,omitempty" protobuf:"bytes,12,opt,name=provider,casttype=TokenProviderName"`
-	GoogleOAuth *GoogleOAuthProvider `json:"googleOAuth,omitempty" protobuf:"bytes,13,opt,name=googleOAuth"`
-	AWS         *AWSProvider         `json:"aws,omitempty" protobuf:"bytes,14,opt,name=aws"`
+	Provider       TokenProviderName      `json:"provider,omitempty" protobuf:"bytes,12,opt,name=provider,casttype=TokenProviderName"`
+	GoogleOAuth    *GoogleOAuthProvider   `json:"googleOAuth,omitempty" protobuf:"bytes,13,opt,name=googleOAuth"`
+	AWS            *AWSProvider           `json:"aws,omitempty" protobuf:"bytes,14,opt,name=aws"`
+	CredentialName string                 `json:"credentialName,omitempty" protobuf:"bytes,15,opt,name=credentialName"`
```
Comment (masudur-rahman) [inline apis/cluster/v1alpha1/cluster_user_auth_types.go:90]: and the credential name in the last

**Review by masudur-rahman — CHANGES_REQUESTED** [review]

File: apis/cluster/v1alpha1/cluster_user_auth_types.go (hunk 1 above)
Comment (Shaad7) [reply apis/cluster/v1alpha1/cluster_user_auth_types.go]: Updated

File: apis/cluster/v1alpha1/cluster_user_auth_types.go (hunk 2 above)
Comment (Shaad7) [reply apis/cluster/v1alpha1/cluster_user_auth_types.go]: Updated

**Review by masudur-rahman — APPROVED** [review]

**Pull request merged**